public/scripts/extensions/translate/index.js

```diff
@@ -582,19 +604,53 @@ async function translateMessageEdit(messageId) {
     }
 }
 
+async function translateMessageReasoningEdit(messageId) {
+    const context = getContext();
+    const chat = context.chat;
+    const message = chat[messageId];
+
+    if (message.is_system || extension_settings.translate.auto_mode == autoModeOptions.NONE) {
+        return;
+    }
+
+    if ((message.is_user && shouldTranslate(outgoingTypes)) || (!message.is_user && shouldTranslate(incomingTypes))) {
+        await translateIncomingMessageReasoning(messageId);
+        context.saveChat();
```

Comment:
```suggestion
        await context.saveChat();
```
Sadly no jsdocs on most of the `getContext` stuff, which can be easy to miss for functions like this, but saving chat is async, so you should await it here.

Reply:
Fixed

Reply:
> Sadly no jsdocs on most of the getContext stuff

Not,  I don't think that's not true, the Promise type is preserved with getContext. But I think we need a Eslint warning for non-awaited promises